### PR TITLE
Retrieve test runs

### PR DESCRIPTION
**Pull request by JasonRitz**
Right now the testCycleId is hard coded. We will need to get that off the user once the view testCycles branch is merged. Also, test coverage for this branch is not that great, but I don't want to hold anybody up so I'll be adding those later. the only changes from that will be in the test files

### CODE REVIEW

File: Mobile Test Runner for Jama/Mobile Test Runner for Jama/TestListViewController.swift [hunk 1]
```diff
@@ -110,7 +133,21 @@ extension TestListViewController: EndpointDelegate {
...
+                    if tmpList.testRunList.isEmpty {
+                        //TODO: display message for no test runs
+                        return
+                    }
+                    self.testRunList.testRunList.append(contentsOf: tmpList.testRunList)
+                    
+                    //TODO: once the testRuns view is made we need to reload the data in the view
+                    //reload Data in view?
+                    
+                    //keep calling api while there are still more cycles
```
Comment (benleelawrence) [inline Mobile Test Runner for Jama/Mobile Test Runner for Jama/TestListViewController.swift]: Should this be still more tests?

File: Mobile Test Runner for Jama/Mobile Test Runner for Jama/TestListViewController.swift [hunk 2]
```diff
@@ -140,12 +173,25 @@ extension TestListViewController: UITableViewDelegate, UITableViewDataSource {
...
+        //Unselect Cycle
+        if indexPath.row - selectedPlanIndex - 1 == selectedCycleIndex {
+            testRunList.testRunList = []
+            selectedCycleIndex = Int.max
+            testList.reloadData()
+        }
         //tapped on a cycle do something
-        if indexPath.row > selectedPlanIndex && indexPath.row <= selectedPlanIndex + totalCyclesVisable {
+        if indexPath.row > selectedPlanIndex && indexPath.row <= selectedPlanIndex + totalCyclesVisible {
+            selectedCycleIndex = indexPath.row <= selectedCycleIndex ? indexPath.row - selectedPlanIndex - 1 : indexPath.row - selectedPlanIndex - 1 // TODO: should be - totalRunsVisable add that in after we view the runs. we are not currently loading in the runs
```
Comment (benleelawrence) [inline Mobile Test Runner for Jama/Mobile Test Runner for Jama/TestListViewController.swift]: Visible misspelled

File: Mobile Test Runner for Jama/Mobile Test Runner for Jama/TestListViewController.swift [hunk 3]
```diff
@@ -140,12 +173,25 @@ extension TestListViewController: UITableViewDelegate, UITableViewDataSource {
...
+            testRunList.testRunList = []
+            testCycleId = testCycleList.testCycleList[selectedCycleIndex].id
+            getRunsForCycleOnClick()
+            
+            
             return
         }
         //TODO will need to subtract totalRunsVisable also
```
Comment (benleelawrence) [inline Mobile Test Runner for Jama/Mobile Test Runner for Jama/TestListViewController.swift]: Visible misspelled

Comment (benleelawrence) [inline Mobile Test Runner for Jama/Mobile Test Runner for Jama/TestListViewController.swift]: Visible misspelled

File: Mobile Test Runner for Jama/Mobile Test Runner for JamaTests/TestListViewControllerUnitTests.swift [hunk 4]
```diff
@@ -11,6 +11,7 @@ import XCTest
 class TestListViewControllerUnitTests: XCTestCase {
     let projectId = 23314
     let planId = 31132
+    let testCycleId = 6835
     var viewController : TestListViewController!
     let instance = "test-instance"
```
Comment (benleelawrence) [inline Mobile Test Runner for Jama/Mobile Test Runner for JamaTests/TestListViewControllerUnitTests.swift:17]: Visible is misspelled in this file on line 42 causing a build failure

**Review by benleelawrence — APPROVED** [review]
Code looks good, builds, all tests pass, great job guys

File: Mobile Test Runner for Jama/Mobile Test Runner for Jama/TestListViewController.swift [hunk 5]
```diff
@@ -110,7 +132,18 @@ extension TestListViewController: EndpointDelegate {
...
-                    let _ = 0  //TO DO: Add call for test runs here and remove this line
+                    let tmpList = TestRunListModel()
+                    tmpList.extractRunList(fromData: unwrappedData, parentId: self.testCycleId)
+                    if tmpList.testRunList.isEmpty {
+                        return
+                    }
+                    self.testRunList.testRunList.append(contentsOf: tmpList.testRunList)
+                    self.totalRunsVisible = self.testRunList.testRunList.count
+                    self.testList.reloadData()
+                                       //keep calling api while there are still more cycles
```
Comment (lcc4) [inline Mobile Test Runner for Jama/Mobile Test Runner for Jama/TestListViewController.swift]: Are there more cycles or more test runs?

File: Mobile Test Runner for Jama/Mobile Test Runner for Jama/TestListViewController.swift [hunk 6]
```diff
@@ -140,12 +173,25 @@ extension TestListViewController: UITableViewDelegate, UITableViewDataSource {
...
+            testRunList.testRunList = []
+            testCycleId = testCycleList.testCycleList[selectedCycleIndex].id
+            getRunsForCycleOnClick()
+            
+            
             return
         }
-        //TODO will need to subtract totalRunsVisable also
-        selectedPlanIndex = indexPath.row <= selectedPlanIndex ? indexPath.row : indexPath.row - totalCyclesVisable
+        //TODO will need to subtract totalRunsVisible also
```
Comment (lcc4) [inline Mobile Test Runner for Jama/Mobile Test Runner for Jama/TestListViewController.swift:193]: Does the subtract totalRunsVisible need to happen still?

**Pull request merged**